shopfloor/actions/stock.py

```diff
@@ -237,3 +238,20 @@ def no_putaway_available(self, picking_types, move_lines):
         # when no putaway is found, the move line destination stays the
         # default's of the picking type
         return any(line.location_dest_id in base_locations for line in move_lines)
+
+    def _lock_lines(self, lines):
+        self._actions_for("lock").for_update(lines)
+
+    def _set_destination_on_lines(self, lines, location_dest):
+        # when writing the destination on the package level, it writes
+        # on the moves and move lines
+        lines.picking_id.location_dest_id = location_dest
```

Comment:
It doesn't look right to change the location of the picking and all its moves. Only the lines should be updated
```suggestion
    lines.location_dest_id = location_dest
    lines.package_level_id.location_dest_id = location_dest
```

Reply:
Yes you are right, i just restored as it is done in v14's PR.